components/eamxx/src/share/field/field_impl.hpp

```diff
@@ -222,6 +222,198 @@ get_strided_view_type<DT, HD> {
   return DstView(get_ND_view<HD, DstValueType, DstRank>());
 }
 
+template<typename ST>
```

Comment:
Up to you, but since the host->dev and dev->host versions only diff b/c of which one is src/dst, maybe you could just have one impl:
```c++
template<typename ST, HostOrDevice From, HostOrDevice To>
void Field sync_views_impl ();
```
And save ~100 lines of code.